src/nemos/tree_utils.py

```diff
@@ -140,3 +141,65 @@ def pytree_map_and_reduce(
     cond_tree = jax.tree_util.tree_map(map_fn, *pytrees, is_leaf=is_leaf)
     # for some reason, tree_reduce doesn't work well with any.
     return reduce_fn(jax.tree_util.tree_leaves(cond_tree))
+
+
+def tree_slice(data: Any, idx):
```

Comment:
I thought slicing already worked on our pytrees, why was this necessary?

Reply:
It's for slicing any pytree, not just `FeaturePytree`.